### PR TITLE
Add the debugOnFail option (closes #1608)

### LINKED ISSUE #1608: Add an option to enable debug mode if a test is failed
I suppose it would be useful if test run will stop and be switched to debug mode when a test is failed to see what happened on the page. It also would be nice to see the error in this moment.
Something like `--debug-on-fail` or `--pause-on-fail`
UPD: when test is stopped the page should be 'unfrozen'. It should be possible to perform interaction with the page elements.

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
Here is how does it look now: https://www.screencast.com/t/kDkx9J5sj

/cc @helen-dikareva @AndreyBelym @VasilyStrelyaev @DevExpress/testcafe

### CODE REVIEW

**Comment by helen-dikareva** [comment]
1) Do we need first action `debug()` in test when we use this option? I think we don't.
2) After `click('nothing') test fails and we can see title `Test failed` but I think we should change button title to `Finish test run` or smth like this because we will not resume test we just finish test run (`Resume` looks like we can skip failsed action).

**Comment by AlexanderMoskovkin** [comment]
> Do we need first action debug() in test when we use this option? I think we don't.

No, we don't need. I've added this to the screencast just for demonstration. If it were no `debug()` commands in the test it'll be stopped when it's failed as well.

> After click('nothing') test fails and we can see title Test failed but I think we should change button title to Finish test run or smth like this because we will not resume test we just finish test run (Resume looks like we can skip failsed action).

It makes sense. We can write just `Finish`, what do you think? /cc @helen-dikareva @VasilyStrelyaev

**Comment by helen-dikareva** [comment]
> No, we don't need.

Ok, it means I understood it right.

> We can write just Finish

Looks good for me

**Comment by VasilyStrelyaev** [comment]
`Finish` looks good

**Comment by AlexanderMoskovkin** [comment]
FPR https://www.screencast.com/t/aasT445be6G

File: src/cli/argument-parser.js [hunk 1]
```diff
@@ -99,6 +99,7 @@ export default class CLIArgumentParser {
             .option('-F, --fixture-grep <pattern>', 'run only fixtures matching the specified pattern')
             .option('-a, --app <command>', 'launch the tested app using the specified command before running tests')
             .option('-c, --concurrency <number>', 'run tests concurrently')
+            .option('--debug-on-fail', 'pause the test if it is failed')
```
Comment (VasilyStrelyaev) [inline src/cli/argument-parser.js]:
```text
            .option('--debug-on-fail', 'pause the test if it fails')
```

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
@kirovboris Yep, I'll add a server test for CLI arguments parsing. But we can't add a functional test for this

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

File: src/test-run/index.js [hunk 2]
```diff
@@ -91,7 +93,9 @@ export default class TestRun extends Session {
         this.resolveWaitForFileDownloadingPromise = null;
 
         this.debugging               = this.opts.debugMode;
+        this.debugOnFail             = this.opts.debugOnFail;
         this.disableDebugBreakpoints = false;
+        this.debugReporterPluginHost = new ReporterPluginHost(testcafeReporterSpecPlugin);
```
Comment (AndreyBelym) [inline src/test-run/index.js]: Requiring `spec` reporter only for using `formatError` helper seems like a bit overkill for me. Personally I would prefer to use `new ReporterPluginHost(null)` or `new ReporterPluginHost()`, maybe with comment that we aren't going to actually use reporter plugin, and only want to use formatting helpers. Or maybe it's even better to extract formatting helpers to a separate class like a `Formatter`.

Comment (AlexanderMoskovkin) [reply src/test-run/index.js]: nice point

**Comment by AlexanderMoskovkin** [comment]
FPR

**Pull request merged**